### DISCUSSION TITLE
About app version

### DISCUSSION THREAD

**Discussion by KaranKad** [microsoft/winget-pkgs#13862] [Ideas]
Some apps have long release number on their website and in about section of the app, but when installed, control panel/settings app show the release in short version.
### In about section and website
![image](https://user-images.githubusercontent.com/71691514/119223343-7b76f400-bb16-11eb-81ab-28d79af0d464.png) 
![image](https://user-images.githubusercontent.com/71691514/119223483-5a62d300-bb17-11eb-8dd0-5d209260e021.png)

### In settings app
![image](https://user-images.githubusercontent.com/71691514/119223439-11ab1a00-bb17-11eb-8804-dbc63994d33d.png)

If I keep the manifest according to the app and website, winget thinks the app is outdated because it checks the app version according to control panel/settings app and tries to install it.

I have latest version of CPU-Z v1.96.1 installed but the app in control panel reports as 1.96, so winget thinks its outdated and tries to upgrade the app.
### In about section
![image](https://user-images.githubusercontent.com/71691514/119223646-3358d100-bb18-11eb-9c33-de4b1fe7b5a3.png)
### winget upgrade command showing app has update
![image](https://user-images.githubusercontent.com/71691514/119223705-8763b580-bb18-11eb-9e9a-5d023a901407.png)
### In settings app
![image](https://user-images.githubusercontent.com/71691514/119223770-d3165f00-bb18-11eb-997a-920f86a584bf.png)

### Some apps don't have version number at all
![image](https://user-images.githubusercontent.com/71691514/119223817-1ffa3580-bb19-11eb-8882-7d57556090fe.png)
![image](https://user-images.githubusercontent.com/71691514/119223824-2dafbb00-bb19-11eb-92ae-629abb883706.png)

This would be a non issue if apps reported their version properly, but sadly they don't.

**Comment by OfficialEsco** [comment]
Since `winget upgrade` just got out of experimental there is a few issues, WinGet is looking at the version which is shown in the Control Panel/Settings App (Which is the version you should type in the Version field)

Main issue thread: https://github.com/microsoft/winget-cli/issues/976
Feature Request for the bot: https://github.com/microsoft/winget-pkgs/issues/13620
Alternative Version Feature Request: https://github.com/microsoft/winget-cli/issues/980

I also see that `winget upgrade` is trying to upgrade to `latest`, which IMO it should not try to do

**Comment by KaranKad** [comment]
> Control Panel/Settings App (Which is the version you should type in the Version field)

I will update manifests according to it then.

**Comment by jedieaston** [comment]
> I also see that `winget upgrade` is trying to upgrade to `latest`, which IMO it should not try to do

It tries to do this for Spotify too, which is rather annoying. At least MSIs are upgraded right now which is nice.

**Comment by denelon** [comment]
We're tracking Issues in the client repository where we think logic changes might help. I think I've seen most of these reported on https://github.com/microsoft/winget-cli/issues/976. I'm going to go ahead and convert this to a discussion, so it isn't closed as an issue.

The current experimental store source doesn't report versions for packages. All we have is "latest" as a sentinel value. That results in store packages always thinking there is an upgrade available. Packages installed from the store generally get automatically updated as long as no policy restricts that behavior. 

I'm interested in more discussion here around maybe changing to version "unknown" or something like that so the behavior could make more sense. Maybe we need an option for upgrade to try upgrading things marked "latest" or "unknown".

Spotify is behind a vanity URL and the installer changes frequently. It's been quite a challenge to keep the right hash on the manifest. We've added a couple of features to try and help. https://github.com/microsoft/winget-cli/issues/870 and https://github.com/microsoft/winget-cli/issues/920.
